### PR TITLE
SNOW-1432694 Add documentation pages for Index object

### PR DESCRIPTION
**Pull request by sfc-gh-vbudati**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1432694

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Adding one page with a list of the Index APIs (indexing.rst) and one page with a table of the APIs supported/not supported. (index_supported.rst)
   This should be merged after the [NotImplementedError PR](https://github.com/snowflakedb/snowpark-python/pull/1750) for Index since all Index APIs are listed here.

### CODE REVIEW

**Comment by sfc-gh-joshi** [comment]
Are you able to build without warnings? After deleting pandas_api and running make clean + make html, I get these warnings:
```
WARNING: [autosummary] failed to import snowflake.snowpark.modin.pandas.Index.T.
Possible hints:
* ImportError: 
* ModuleNotFoundError: No module named 'snowflake.snowpark.modin.pandas.Index'
* AttributeError: type object 'Index' has no attribute 'T'
WARNING: [autosummary] failed to import snowflake.snowpark.modin.pandas.Index.all.
Possible hints:
* ImportError: 
* AttributeError: type object 'Index' has no attribute 'all'
* ModuleNotFoundError: No module named 'snowflake.snowpark.modin.pandas.Index'
```
(repeated many times for other methods)

```
checking consistency... /Users/joshi/code/snowpandas/docs/source/modin/supported/index_supported.rst: WARNING: document isn't included in any toctree
```

```
/Users/joshi/code/snowpandas/src/snowflake/snowpark/modin/plugin/_internal/index.py:docstring of snowflake.snowpark.modin.plugin._internal.index.Index:7: WARNING: undefined label: basics.dtypes
```

**Comment by sfc-gh-vbudati** [comment]
@sfc-gh-joshi It will keep on raising the build errors until my other PR is in (raising NotImplementedError for all APIs not present). It's raising the errors because these APIs don't exist in the code.

**Review by sfc-gh-nkrishna — COMMENTED** [review]
@sfc-gh-samdesai1 Could you PTAL at this as well?

**Review by sfc-gh-azhan — COMMENTED** [review]
let's update change log too

File: src/snowflake/snowpark/modin/plugin/_internal/index.py [hunk 1]
```diff
@@ -177,6 +177,8 @@ def __getattr__(self, key: str) -> Any:
             if not key.startswith("_"):
                 native_index = native_pd.Index([])
                 if hasattr(native_index, key):
+                    # Any methods that not supported by the current Index.py but exist in a
```
Comment (sfc-gh-vbudati) [inline src/snowflake/snowpark/modin/plugin/_internal/index.py:180]: comment Yun requested in https://github.com/snowflakedb/snowpark-python/pull/1751

**Review by sfc-gh-samdesai1 — APPROVED** [review]
Looks good!

**Review by sfc-gh-nkrishna — APPROVED** [review]
LGTM!

**Review by sfc-gh-azhan — APPROVED** [review]

**Pull request merged**